Build failure: liblcf

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Try to build `easyrpg-player`
2. Fails to build `liblcf`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/290636861

### Relevant log output

```console
Running phase: unpackPhase
unpacking source archive /nix/store/7h29qh4n85ap6bdx4vv5lcgw918ihn42-source
source root is source
Running phase: patchPhase
Running phase: autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force --install -I builds/autoconf/m4
aclocal: installing 'builds/autoconf/m4/pkg.m4' from '/nix/store/601zbsyxn1zqx22yrif9yxa1x450qfc8-pkg-config-wrapper-0.29.2/share/aclocal/pkg.m4'
aclocal: installing 'builds/autoconf/m4/libtool.m4' from '/nix/store/iadsrgl73xp5kcf0r5x03nrlym7xss3i-libtool-2.5.4/share/aclocal/libtool.m4'
aclocal: installing 'builds/autoconf/m4/ltoptions.m4' from '/nix/store/iadsrgl73xp5kcf0r5x03nrlym7xss3i-libtool-2.5.4/share/aclocal/ltoptions.m4'
aclocal: installing 'builds/autoconf/m4/ltsugar.m4' from '/nix/store/iadsrgl73xp5kcf0r5x03nrlym7xss3i-libtool-2.5.4/share/aclocal/ltsugar.m4'
aclocal: installing 'builds/autoconf/m4/ltversion.m4' from '/nix/store/iadsrgl73xp5kcf0r5x03nrlym7xss3i-libtool-2.5.4/share/aclocal/ltversion.m4'
aclocal: installing 'builds/autoconf/m4/lt~obsolete.m4' from '/nix/store/iadsrgl73xp5kcf0r5x03nrlym7xss3i-libtool-2.5.4/share/aclocal/lt~obsolete.m4'
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'builds/autoconf'.
libtoolize: copying file 'builds/autoconf/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'builds/autoconf/m4'.
libtoolize: copying file 'builds/autoconf/m4/libtool.m4'
libtoolize: copying file 'builds/autoconf/m4/ltoptions.m4'
libtoolize: copying file 'builds/autoconf/m4/ltsugar.m4'
libtoolize: copying file 'builds/autoconf/m4/ltversion.m4'
libtoolize: copying file 'builds/autoconf/m4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force --install -I builds/autoconf/m4
autoreconf: running: /nix/store/m9qpqr7m6ik1szwlrnk2sjw7x7k0rdbg-autoconf-2.72/bin/autoconf --force
configure.ac:80: warning: AC_C_BIGENDIAN should be used with AC_CONFIG_HEADERS
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:17: installing 'builds/autoconf/ar-lib'
configure.ac:17: installing 'builds/autoconf/compile'
configure.ac:20: installing 'builds/autoconf/config.guess'
configure.ac:20: installing 'builds/autoconf/config.sub'
configure.ac:8: installing 'builds/autoconf/install-sh'
configure.ac:8: installing 'builds/autoconf/missing'
Makefile.am: installing 'builds/autoconf/depcomp'
autoreconf: 'builds/autoconf/install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./builds/autoconf/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./builds/autoconf/config.guess
Running phase: configurePhase
fixing libtool script ./builds/autoconf/ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/16yp9jqvxzdpaqkgzg173v4cd8bbm9m8-liblcf-0.8
checking for a BSD-compatible install... /nix/store/fr9yr63a1267cr0r86w18c77mh3xavcc-coreutils-9.6/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/fr9yr63a1267cr0r86w18c77mh3xavcc-coreutils-9.6/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking the archiver (ar) interface... ar
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... none
checking how to run the C++ preprocessor... g++ -E
checking whether the compiler supports GNU C++... (cached) yes
checking whether g++ accepts -g... (cached) yes
checking for g++ option to enable C++11 features... (cached) none needed
checking dependency style of g++... (cached) none
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/fcny04vmfnkqc4ym5m9znckswlq9pglg-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/yz6nq9v29l3g990w8zlqsnzlzbicd3b3-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/yz6nq9v29l3g990w8zlqsnzlzbicd3b3-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/yz6nq9v29l3g990w8zlqsnzlzbicd3b3-gnugrep-3.11/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ranlib... ranlib
checking for archiver @FILE support... @
checking for strip... strip
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/fr9yr63a1267cr0r86w18c77mh3xavcc-coreutils-9.6/bin/dd
checking how to truncate binary pipes... /nix/store/fr9yr63a1267cr0r86w18c77mh3xavcc-coreutils-9.6/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking pkg-config is at least version 0.9.0... yes
checking for  icu-i18n... yes
checking for  expat >= 2.1... yes
checking for update-mime-database... no
configure: WARNING: Could not find the update-mime-database program in your PATH. Your mime database will not be updated upon install.
checking for stdint.h... (cached) yes
checking for _Bool... no
checking for stdbool.h that conforms to C99 or later... yes
checking for int16_t... yes
checking for int32_t... yes
checking for size_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint8_t... yes
checking whether byte ordering is bigendian... no
checking for floor... yes
checking for strrchr... yes
checking for strtol... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating builds/liblcf.pc
config.status: creating builds/autoconf/liblcf-config.cmake
config.status: creating builds/autoconf/liblcf-config-version.cmake
config.status: creating src/lcf/config.h
config.status: executing depfiles commands
config.status: executing libtool commands
Running phase: buildPhase
build flags: -j24 SHELL=/nix/store/11ciq72n4fdv8rw6wgjgasfv4mjs1jrw-bash-5.2p37/bin/bash
  CXX      tools/lcf2xml-lcf2xml.o
  CXX      tools/lcfstrings-lcfstrings.o
  CXX      src/liblcf_la-dbarray.lo
  CXX      src/liblcf_la-dbstring_struct.lo
  CXX      src/liblcf_la-encoder.lo
  CXX      src/liblcf_la-ini.lo
  CXX      src/liblcf_la-inireader.lo
  CXX      src/liblcf_la-ldb_equipment.lo
  CXX      src/liblcf_la-ldb_eventcommand.lo
  CXX      src/liblcf_la-ldb_parameters.lo
  CXX      src/liblcf_la-ldb_reader.lo
  CXX      src/liblcf_la-lmt_reader.lo
  CXX      src/liblcf_la-lmt_rect.lo
  CXX      src/liblcf_la-lmt_treemap.lo
  CXX      src/liblcf_la-lmu_movecommand.lo
  CXX      src/liblcf_la-lmu_reader.lo
  CXX      src/liblcf_la-lsd_reader.lo
  CXX      src/liblcf_la-reader_flags.lo
  CXX      src/liblcf_la-reader_lcf.lo
  CXX      src/liblcf_la-reader_util.lo
  CXX      src/liblcf_la-reader_xml.lo
  CXX      src/liblcf_la-rpg_setup.lo
  CXX      src/liblcf_la-rpg_terms.lo
  CXX      src/liblcf_la-saveopt.lo
  CXX      src/liblcf_la-writer_lcf.lo
In file included from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uenum.h:25,
                 from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucsdet.h:28,
                 from src/reader_util.cpp:14:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:561:26: error: 'auto' parameter not permitted in this context
  561 | template <typename Type, auto closeFunction>
      |                          ^~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:573:76: error: template argument 2 is invalid
  573 |     explicit LocalOpenPointer(std::unique_ptr<Type, decltype(closeFunction)> &&p)
      |                                                                            ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:583:78: error: template argument 2 is invalid
  583 |     LocalOpenPointer &operator=(std::unique_ptr<Type, decltype(closeFunction)> &&p) {
      |                                                                              ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:599:59: error: template argument 2 is invalid
  599 |     operator std::unique_ptr<Type, decltype(closeFunction)> () && {
      |                                                           ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uenum.h:69:1: note: invalid template non-type parameter
   69 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUEnumerationPointer, UEnumeration, uenum_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucsdet.h:109:1: note: invalid template non-type parameter
  109 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUCharsetDetectorPointer, UCharsetDetector, ucsdet_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucnv.h:597:1: note: invalid template non-type parameter
  597 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUConverterPointer, UConverter, ucnv_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/normalizer2.h:33,
                 from src/reader_util.cpp:16:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/stringpiece.h:135:23: error: 'is_same_v' is not a member of 'std'; did you mean 'is_same'?
  135 |                 (std::is_same_v<decltype(T().data()), const char*>
      |                       ^~~~~~~~~
      |                       is_same
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/stringpiece.h:135:66: error: template argument 1 is invalid
  135 |                 (std::is_same_v<decltype(T().data()), const char*>
      |                                                                  ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/stringpiece.h:139:17: error: expected '>' before ')' token
  139 |                 ) &&
      |                 ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/stringpiece.h:139:17: error: expected unqualified-id before ')' token
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/stringpiece.h:185:19: error: expected type-specifier
  185 |   inline operator std::string_view() const {
      |                   ^~~
In file included from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:37,
                 from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uniset.h:22,
                 from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/normalizer2.h:34:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:317:10: error: 'is_convertible_v' is not a member of 'std'; did you mean 'is_convertible'?
  317 |     std::is_convertible_v<T, std::u16string_view>
      |          ^~~~~~~~~~~~~~~~
      |          is_convertible
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:317:28: error: expected primary-expression before ',' token
  317 |     std::is_convertible_v<T, std::u16string_view>
      |                            ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:331:13: error: 'u16string_view' in namespace 'std' does not name a type; did you mean 'u16string'?
  331 | inline std::u16string_view toU16StringView(std::u16string_view sv) { return sv; }
      |             ^~~~~~~~~~~~~~
      |             u16string
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:339:13: error: 'u16string_view' in namespace 'std' does not name a type; did you mean 'u16string'?
  339 | inline std::u16string_view toU16StringView(std::basic_string_view<uint16_t> sv) {
      |             ^~~~~~~~~~~~~~
      |             u16string
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:360:54: error: 'is_pointer_v' is not a member of 'std'; did you mean 'is_pointer'?
  360 |           typename = typename std::enable_if_t<!std::is_pointer_v<std::remove_reference_t<T>>>>
      |                                                      ^~~~~~~~~~~~
      |                                                      is_pointer
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:360:92: error: template argument 1 is invalid
  360 |           typename = typename std::enable_if_t<!std::is_pointer_v<std::remove_reference_t<T>>>>
      |                                                                                            ^~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:360:94: error: expected identifier before '>>' token
  360 |           typename = typename std::enable_if_t<!std::is_pointer_v<std::remove_reference_t<T>>>>
      |                                                                                              ^~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:360:94: error: expected '>' before '>>' token
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:361:13: error: 'u16string_view' in namespace 'std' does not name a type; did you mean 'u16string'?
  361 | inline std::u16string_view toU16StringViewNullable(const T& text) {
      |             ^~~~~~~~~~~~~~
      |             u16string
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:370:53: error: 'is_pointer_v' is not a member of 'std'; did you mean 'is_pointer'?
  370 |           typename = typename std::enable_if_t<std::is_pointer_v<std::remove_reference_t<T>>>,
      |                                                     ^~~~~~~~~~~~
      |                                                     is_pointer
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:370:53: error: 'is_pointer_v' is not a member of 'std'; did you mean 'is_pointer'?
  370 |           typename = typename std::enable_if_t<std::is_pointer_v<std::remove_reference_t<T>>>,
      |                                                     ^~~~~~~~~~~~
      |                                                     is_pointer
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:370:91: error: template argument 1 is invalid
  370 |           typename = typename std::enable_if_t<std::is_pointer_v<std::remove_reference_t<T>>>,
      |                                                                                           ^~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:370:93: error: expected identifier before '>' token
  370 |           typename = typename std::enable_if_t<std::is_pointer_v<std::remove_reference_t<T>>>,
      |                                                                                             ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/char16ptr.h:370:94: error: expected unqualified-id before ',' token
  370 |           typename = typename std::enable_if_t<std::is_pointer_v<std::remove_reference_t<T>>>,
      |                                                                                              ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:3035:19: error: expected type-specifier
 3035 |   inline operator std::u16string_view() const {
      |                   ^~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:3755:60: error: 'std::u16string_view' has not been declared
 3755 |   static UnicodeString readOnlyAliasFromU16StringView(std::u16string_view text);
      |                                                            ^~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:3871:64: error: 'std::u16string_view' has not been declared
 3871 |   UnicodeString& doReplace(int32_t start, int32_t length, std::u16string_view src);
      |                                                                ^~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:3875:32: error: 'std::u16string_view' has not been declared
 3875 |   UnicodeString& doAppend(std::u16string_view src);
      |                                ^~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In member function 'bool icu_76::UnicodeString::operator==(const S&) const':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:348:10: error: 'u16string_view' is not a member of 'std'; did you mean 'u16string'?
  348 |     std::u16string_view sv(internal::toU16StringView(text));
      |          ^~~~~~~~~~~~~~
      |          u16string
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:350:46: error: 'sv' was not declared in this scope
  350 |     return !isBogus() && (len = length()) == sv.length() && doEquals(sv.data(), len);
      |                                              ^~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In member function 'icu_76::UnicodeString& icu_76::UnicodeString::operator=(const S&)':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:1962:45: error: 'toU16StringView' is not a member of 'icu_76::internal'
 1962 |     return doReplace(0, length(), internal::toU16StringView(src));
      |                                             ^~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In member function 'icu_76::UnicodeString& icu_76::UnicodeString::operator+=(const S&)':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:2228:31: error: 'toU16StringView' is not a member of 'icu_76::internal'
 2228 |     return doAppend(internal::toU16StringView(src));
      |                               ^~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In member function 'icu_76::UnicodeString& icu_76::UnicodeString::append(const S&)':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:2301:31: error: 'toU16StringView' is not a member of 'icu_76::internal'
 2301 |     return doAppend(internal::toU16StringView(src));
      |                               ^~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In constructor 'icu_76::UnicodeString::UnicodeString(const S&)':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:3276:24: error: 'toU16StringViewNullable' is not a member of 'icu_76::internal'
 3276 |     doAppend(internal::toU16StringViewNullable(text));
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In static member function 'static icu_76::UnicodeString icu_76::UnicodeString::readOnlyAlias(const S&)':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:3601:53: error: 'toU16StringView' is not a member of 'icu_76::internal'
 3601 |     return readOnlyAliasFromU16StringView(internal::toU16StringView(text));
      |                                                     ^~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: In function 'icu_76::UnicodeString icu_76::operator+(const UnicodeString&, const S&)':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:4118:46: error: 'toU16StringView' is not a member of 'icu_76::internal'
 4118 |   return unistr_internalConcat(s1, internal::toU16StringView(s2));
      |                                              ^~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:4118:10: error: there are no arguments to 'unistr_internalConcat' that depend on a template parameter, so a declaration of 'unistr_internalConcat' must be available [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;]
 4118 |   return unistr_internalConcat(s1, internal::toU16StringView(s2));
      |          ^~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:4118:10: note: (if you use '8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h: At global scope:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unistr.h:4125:53: error: 'std::u16string_view' has not been declared
 4125 | unistr_internalConcat(const UnicodeString &s1, std::u16string_view s2);
      |                                                     ^~~~~~~~~~~~~~
In file included from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uenum.h:25,
                 from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucsdet.h:28,
                 from src/encoder.cpp:19:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:561:26: error: 'auto' parameter not permitted in this context
  561 | template <typename Type, auto closeFunction>
      |                          ^~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:573:76: error: template argument 2 is invalid
  573 |     explicit LocalOpenPointer(std::unique_ptr<Type, decltype(closeFunction)> &&p)
      |                                                                            ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:583:78: error: template argument 2 is invalid
  583 |     LocalOpenPointer &operator=(std::unique_ptr<Type, decltype(closeFunction)> &&p) {
      |                                                                              ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/localpointer.h:599:59: error: template argument 2 is invalid
  599 |     operator std::unique_ptr<Type, decltype(closeFunction)> () && {
      |                                                           ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uenum.h:69:1: note: invalid template non-type parameter
   69 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUEnumerationPointer, UEnumeration, uenum_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uset.h:361:1: note: invalid template non-type parameter
  361 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUSetPointer, USet, uset_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucsdet.h:109:1: note: invalid template non-type parameter
  109 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUCharsetDetectorPointer, UCharsetDetector, ucsdet_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucnv.h:597:1: note: invalid template non-type parameter
  597 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUConverterPointer, UConverter, ucnv_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uniset.h:23:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uset.h:1655:10: error: 'u16string_view' in namespace 'std' does not name a type; did you mean 'u16string'?
 1655 |     std::u16string_view operator*() const {
      |          ^~~~~~~~~~~~~~
      |          u16string
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/unorm2.h:284:1: note: invalid template non-type parameter
  284 | U_DEFINE_LOCAL_OPEN_POINTER(LocalUNormalizer2Pointer, UNormalizer2, unorm2_close);
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:1958: src/liblcf_la-encoder.lo] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:2063: src/liblcf_la-reader_util.lo] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.3, NixOS, 25.05 (Warbler), 25.05.20250221.32fb99b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/qc7n2v5yl42m2wq59f59ly6igiwljh11-source`

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/emulators/libretro/cores/easyrpg.nix
+++ b/pkgs/applications/emulators/libretro/cores/easyrpg.nix
@@ -1,64 +1,73 @@
 {
   lib,
   fetchFromGitHub,
+  mkLibretroCore,
+  nix-update-script,
+  asciidoctor,
   cmake,
-  fetchpatch,
+  doxygen,
+  pkg-config,
+  flac,
+  fluidsynth,
   fmt,
   freetype,
+  glib,
   harfbuzz,
+  lhasa,
   liblcf,
   libpng,
   libsndfile,
+  libsysprof-capture,
   libvorbis,
   libxmp,
-  mkLibretroCore,
   mpg123,
+  nlohmann_json,
   opusfile,
-  pcre,
+  pcre2,
   pixman,
-  pkg-config,
   speexdsp,
+  wildmidi,
 }:
-mkLibretroCore {
+mkLibretroCore rec {
   core = "easyrpg";
-  version = "0.8-unstable-2023-04-29";
+  # liblcf needs to be updated before this.
+  version = "0.8.1.1";
 
   src = fetchFromGitHub {
     owner = "EasyRPG";
     repo = "Player";
-    rev = "f8e41f43b619413f95847536412b56f85307d378";
-    hash = "sha256-nvWM4czTv/GxY9raomBEn7dmKBeLtSA9nvjMJxc3Q8s=";
+    rev = version;
+    hash = "sha256-2a8IdYP6Suc8a+Np5G+xoNzuPxkk9gAgR+sjdKUf89M=";
     fetchSubmodules = true;
   };
 
   extraNativeBuildInputs = [
+    asciidoctor
     cmake
+    doxygen
     pkg-config
   ];
   extraBuildInputs = [
+    flac # needed by libsndfile
+    fluidsynth
     fmt
     freetype
+    glib
     harfbuzz
+    lhasa
     liblcf
     libpng
     libsndfile
+    libsysprof-capture # needed by glib
     libvorbis
     libxmp
     mpg123
+    nlohmann_json
     opusfile
-    pcre
+    pcre2 # needed by glib
     pixman
     speexdsp
-  ];
-  patches = [
-    # The following patch is shared with easyrpg-player.
-    # Update when new versions of liblcf and easyrpg-player are released.
-    # See easyrpg-player expression for details.
-    (fetchpatch {
-      name = "0001-Fix-building-with-fmtlib-10.patch";
-      url = "https://github.com/EasyRPG/Player/commit/ab6286f6d01bada649ea52d1f0881dde7db7e0cf.patch";
-      hash = "sha256-GdSdVFEG1OJCdf2ZIzTP+hSrz+ddhTMBvOPjvYQHy54=";
-    })
+    wildmidi
   ];
   cmakeFlags = [
     "-DBUILD_SHARED_LIBS=ON"
@@ -67,12 +76,13 @@ mkLibretroCore {
   ];
   makefile = "Makefile";
 
-  # Do not update automatically since we want to pin a specific version
-  passthru.updateScript = null;
+  # Since liblcf needs to be updated before this, we should not
+  # use the default unstableGitUpdater.
+  passthru.updateScript = nix-update-script { };
 
   meta = {
     description = "EasyRPG Player libretro port";
     homepage = "https://github.com/EasyRPG/Player";
-    license = lib.licenses.gpl3Only;
+    license = lib.licenses.gpl3Plus;
   };
 }

--- a/pkgs/by-name/ea/easyrpg-player/package.nix
+++ b/pkgs/by-name/ea/easyrpg-player/package.nix
@@ -2,59 +2,54 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  fetchpatch,
+  asciidoctor,
   cmake,
   doxygen,
   pkg-config,
-  freetype,
+  alsa-lib,
+  flac,
+  fluidsynth,
   fmt,
+  freetype,
   glib,
   harfbuzz,
+  lhasa,
+  libdecor,
   liblcf,
   libpng,
   libsndfile,
+  libsysprof-capture,
   libvorbis,
-  libxmp,
   libXcursor,
   libXext,
   libXi,
   libXinerama,
+  libxmp,
   libXrandr,
   libXScrnSaver,
   libXxf86vm,
   mpg123,
+  nlohmann_json,
   opusfile,
-  pcre,
+  pcre2,
   pixman,
-  SDL2,
+  sdl3,
   speexdsp,
   wildmidi,
   zlib,
-  libdecor,
-  alsa-lib,
-  asciidoctor,
 }:
 
 stdenv.mkDerivation rec {
   pname = "easyrpg-player";
-  version = "0.8";
+  # liblcf needs to be updated before this.
+  version = "0.8.1.1";
 
   src = fetchFromGitHub {
     owner = "EasyRPG";
     repo = "Player";
     rev = version;
-    hash = "sha256-t0sa9ONVVfsiTy+us06vU2bMa4QmmQeYxU395g0WS6w=";
+    hash = "sha256-fYSpFhqETkQhRK1/Uws0fWWdCr35+1J4vCPX9ZiQ3ZA=";
   };
-
-  patches = [
-    # Fixed compatibility with fmt > 9
-    # Remove when version > 0.8
-    (fetchpatch {
-      name = "0001-Fix-building-with-fmtlib-10.patch";
-      url = "https://github.com/EasyRPG/Player/commit/ab6286f6d01bada649ea52d1f0881dde7db7e0cf.patch";
-      hash = "sha256-GdSdVFEG1OJCdf2ZIzTP+hSrz+ddhTMBvOPjvYQHy54=";
-    })
-  ];
 
   strictDeps = true;
 
@@ -67,21 +62,27 @@ stdenv.mkDerivation rec {
 
   buildInputs =
     [
+      flac # needed by libsndfile
+      fluidsynth
       fmt
       freetype
       glib
       harfbuzz
+      lhasa
       liblcf
       libpng
       libsndfile
+      libsysprof-capture # needed by glib
       libvorbis
       libxmp
       mpg123
+      nlohmann_json
       opusfile
-      pcre
+      pcre2 # needed by glib
       pixman
-      SDL2
+      sdl3
       speexdsp
+      wildmidi
       zlib
     ]
     ++ lib.optionals stdenv.hostPlatform.isLinux [
@@ -94,11 +95,12 @@ stdenv.mkDerivation rec {
       libXScrnSaver
       libXxf86vm
       libdecor
-      wildmidi # until packaged on Darwin
     ];
 
   cmakeFlags = [
     "-DPLAYER_ENABLE_TESTS=${lib.boolToString doCheck}"
+    # TODO: remove the below once SDL3 becomes default next major release
+    "-DPLAYER_TARGET_PLATFORM=SDL3"
   ];
 
   makeFlags = [
@@ -116,14 +118,13 @@ stdenv.mkDerivation rec {
     ln -s $out/{Applications/EasyRPG\ Player.app/Contents/MacOS,bin}/EasyRPG\ Player
   '';
 
-  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
-
   enableParallelChecking = true;
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
 
   meta = with lib; {
     description = "RPG Maker 2000/2003 and EasyRPG games interpreter";
     homepage = "https://easyrpg.org/";
-    license = licenses.gpl3;
+    license = licenses.gpl3Plus;
     maintainers = [ ];
     platforms = platforms.all;
     mainProgram = lib.optionalString stdenv.hostPlatform.isDarwin "EasyRPG Player";

--- a/pkgs/by-name/li/liblcf/package.nix
+++ b/pkgs/by-name/li/liblcf/package.nix
@@ -2,24 +2,28 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  nix-update-script,
   autoreconfHook,
   pkg-config,
   expat,
   icu74,
+  inih,
 }:
 
 stdenv.mkDerivation rec {
   pname = "liblcf";
-  version = "0.8";
+  # When updating this package, you should probably also update
+  # easyrpg-player and libretro.easyrpg
+  version = "0.8.1";
 
   src = fetchFromGitHub {
     owner = "EasyRPG";
     repo = "liblcf";
     rev = version;
-    hash = "sha256-jJGIsNw7wplTL5FBWGL8osb9255o9ZaWgl77R+RLDMM=";
+    hash = "sha256-jIk55+n8wSk3Z3FPR18SE7U3OuWwmp2zJgvSZQBB2l0=";
   };
 
-  dtrictDeps = true;
+  strictDeps = true;
 
   nativeBuildInputs = [
     autoreconfHook
@@ -29,12 +33,15 @@ stdenv.mkDerivation rec {
   propagatedBuildInputs = [
     expat
     icu74
+    inih
   ];
 
   enableParallelBuilding = true;
   enableParallelChecking = true;
 
   doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  passthru.updateScript = nix-update-script { };
 
   meta = with lib; {
     description = "Library to handle RPG Maker 2000/2003 and EasyRPG projects";

